apps/prairielearn/src/components/EditQuestionPointsScore.tsx

```diff
@@ -136,3 +144,298 @@ function EditQuestionPointsScoreForm({
+    e.preventDefault();
+    setError(null);
+
+    const numValue = value === '' ? null : Number.parseFloat(value);
+    if (value !== '' && Number.isNaN(numValue!)) {
+      setError('Please enter a valid number');
+      return;
+    }
+
+    try {
+      const result = await mutation.mutateAsync({
+        instance_question_id: instanceQuestion.id,
+        modified_at: instanceQuestion.modified_at.toISOString(),
+        field,
+        value: numValue,
+      });
+
+      if (result) {
+        onConflict(result.conflict_details_url);
+      } else {
+        onSuccess();
+      }
+    } catch (err) {
+      setError(err instanceof Error ? err.message : 'Failed to update points');
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <div class="mb-3">
+        <div class="input-group">
+          <input
+            type="number"
+            step="any"
+            class="form-control"
+            value={value}
+            aria-label={findLabel(field)}
+            required
+            onChange={(e) => {
+              setValue((e.target as HTMLInputElement).value);
+            }}
+          />
+          <span class="input-group-text">
+            {field === 'score_perc' ? '%' : `/${maxPoints ?? 0}`}
+          </span>
+        </div>
+        {error && <div class="text-danger mt-2">{error}</div>}
+        {mutation.isError && (
+          <div class="text-danger mt-2">
+            {mutation.error.message || 'An error occurred while updating points'}
+          </div>
+        )}
+      </div>
+      <p>
+        <small>
+          This will also recalculate the total points and total score at 100% credit. This change
+          will be overwritten if the question is answered again by the student.
+          {instanceQuestion.status !== 'unanswered' && (
+            <>
+              {' '}
+              You may also update the score{' '}
+              <a href={manualGradingUrl}>via the manual grading page</a>.
+            </>
+          )}
+        </small>
+      </p>
+      <div class="text-end">
+        <Button disabled={mutation.isPending} variant="secondary" onClick={onCancel}>
+          Cancel
+        </Button>
+        <Button disabled={mutation.isPending} type="submit" variant="primary">
+          {mutation.isPending ? 'Changing...' : 'Change'}
+        </Button>
+      </div>
+    </form>
+  );
+}
+
+interface EditQuestionPointsScoreButtonProps {
+  field: 'points' | 'auto_points' | 'manual_points' | 'score_perc';
+  instanceQuestion: StaffInstanceQuestion;
+  assessmentQuestion: StaffAssessmentQuestion;
+  csrfToken: string;
+  urlPrefix: string;
+  onSuccess: () => void;
+  onConflict: (conflictDetailsUrl: string) => void;
+  scrollRef?: React.RefObject<HTMLDivElement> | null;
+}
+
+export function EditQuestionPointsScoreButton({
+  field,
+  instanceQuestion,
+  assessmentQuestion,
+  csrfToken,
+  urlPrefix,
+  onSuccess,
+  onConflict,
+  scrollRef,
+}: EditQuestionPointsScoreButtonProps) {
```

Comment:
Nit: it's been our preference to inline interfaces like this that don't need to be exported or otherwise referred to by name.